src/provider/application.js

```diff
@@ -127,13 +127,16 @@ class Application extends EventEmitter {
         this.JSONRPC.request('authorizeConsumer', [])
           .then(this.authorizeConsumer)
           .catch(this.emitError);
-      }
 
       // 2b. We don't know who to trust, challenge parent for secret
-      if (this.secret) {
+      } else if (this.secret) {
```

Comment:
Was this intentionally not an 'else if' before? I found it odd that you could have an ACL that passed but then a secret that failed.

Reply:
you're right! it should be `else if`. I guess we missed a return statement previously.